src/Adapter/Module/Repository/ModuleRepository.php

```diff
@@ -94,17 +121,78 @@ public function getActiveModulesPaths(): array
     {
         if (null === $this->activeModulesPaths) {
             $this->activeModulesPaths = [];
-            $modulesFiles = Finder::create()->directories()->in(_PS_MODULE_DIR_)->depth(0);
             $activeModules = $this->getActiveModules();
 
-            foreach ($modulesFiles as $moduleFile) {
-                $moduleName = $moduleFile->getFilename();
+            foreach ($this->getModules() as $moduleName => $modulePath) {
                 if (in_array($moduleName, $activeModules)) {
-                    $this->activeModulesPaths[$moduleName] = $moduleFile->getPathname();
+                    $this->activeModulesPaths[$moduleName] = $modulePath;
                 }
             }
         }
 
         return $this->activeModulesPaths;
     }
+
+    /**
+     * Returns an array of native modules
+     *
+     * @return array<string>
+     */
+    public function getNativeModules(): array
+    {
+        // Native modules are the one integrated in PrestaShop release via composer
+        // so we use the lock files to generate the list
+        $content = file_get_contents($this->rootDir . '/composer.lock');
+        $content = json_decode($content, true);
+        if (empty($content['packages'])) {
+            return [];
+        }
+
+        $modules = array_filter($content['packages'], function (array $package) {
+            return self::COMPOSER_PACKAGE_TYPE === $package['type'] && !empty($package['name']);
+        });
+        $modules = array_map(function (array $package) {
+            $vendorName = explode('/', $package['name']);
+
+            return $vendorName[1];
+        }, $modules);
+
+        return array_merge(
+            $modules,
+            static::ADDITIONAL_ALLOWED_MODULES
+        );
+    }
+
+    /**
+     * Returns an array of non-native modules
```

Comment:
```suggestion
     * Returns an array of non-native module names
```

Could be useful to make it clear we are giving module names, not module objects

Reply:
Fixed